test/e2e/performanceprofile/functests/utils/daemonset/daemonset.go

```diff
@@ -46,3 +55,164 @@ func IsRunning(cli client.Client, namespace, name string) (bool, error) {
 	testlog.Infof("daemonset %q %q desired %d scheduled %d ready %d", namespace, name, ds.Status.DesiredNumberScheduled, ds.Status.CurrentNumberScheduled, ds.Status.NumberReady)
 	return (ds.Status.DesiredNumberScheduled > 0 && ds.Status.DesiredNumberScheduled == ds.Status.NumberReady), nil
 }
+
+func WaitForDeletion(cli client.Client, namespace, name string, timeout time.Duration) error {
+	return wait.PollUntilContextTimeout(context.TODO(), time.Second, timeout, true, func(ctx context.Context) (bool, error) {
+		_, err := GetByName(cli, namespace, name)
+		if err != nil {
```

Comment:
you likely want to check this is error is exactly `IsNotFound` (the apierror variant) before to declare success